src/core/src/core_logic/ConfigurePatching.py

```diff
@@ -0,0 +1,72 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Requires Python 2.7+
+
+""" Configure Patching """
+from core.src.bootstrap.Constants import Constants
+
+
+class ConfigurePatching(object):
+    def __init__(self, env_layer, execution_config, composite_logger, telemetry_writer, status_handler, package_manager):
+        self.env_layer = env_layer
+        self.execution_config = execution_config
+
+        self.composite_logger = composite_logger
+        self.telemetry_writer = telemetry_writer
+        self.status_handler = status_handler
+
+        self.package_manager = package_manager
+
+    def start_configure_patching(self):
+        """ Start configure patching """
+        self.status_handler.set_current_operation(Constants.CONFIGURE_PATCHING)
+        self.raise_if_agent_incompatible()
+        self.composite_logger.log('\nStarting configure patching...')
+
+        try:
+            self.status_handler.set_configure_patching_substatus_json(status=Constants.STATUS_TRANSITIONING, automatic_os_patch_state=Constants.PATCH_STATE_UNKNOWN)
+            self.composite_logger.log("\nMachine Id: " + self.env_layer.platform.node())
+            self.composite_logger.log("Activity Id: " + self.execution_config.activity_id)
+            self.composite_logger.log("Operation request time: " + self.execution_config.start_time)
+
+            # get current auto os updates on the machine and log it in status file
+            current_auto_os_patch_state = self.package_manager.get_current_auto_os_patch_state()
+            self.status_handler.set_configure_patching_substatus_json(status=Constants.STATUS_TRANSITIONING, automatic_os_patch_state=current_auto_os_patch_state)
+
+            # disable auto OS updates if VM is configured for platform updates only.
+            if self.execution_config.patch_mode == Constants.AUTOMATIC_BY_PLATFORM and current_auto_os_patch_state == Constants.PATCH_STATE_ENABLED:
+                self.package_manager.disable_auto_os_update()
+
+            # get current auto os updates on the machine and log it in status file
+            current_auto_os_patch_state = self.package_manager.get_current_auto_os_patch_state()
+            self.status_handler.set_configure_patching_substatus_json(status=Constants.STATUS_SUCCESS, automatic_os_patch_state=current_auto_os_patch_state)
+
+        except Exception as error:
+            error_msg = 'Error processing configure patching request: ' + repr(error)
+            self.composite_logger.log_error(error_msg)
+            self.status_handler.add_error_to_status(error_msg, Constants.PatchOperationErrorCodes.DEFAULT_ERROR)
+            if Constants.ERROR_ADDED_TO_STATUS not in repr(error):
+                error.args = (error.args, "[{0}]".format(Constants.ERROR_ADDED_TO_STATUS))
+            self.status_handler.set_configure_patching_substatus_json(status=Constants.STATUS_ERROR)
+            raise
```

Comment:
Review what the Windows extension is doing in this case and expand the common spec accordingly before applying to Linux. The documentation is lacking.

Reply:
ToDo

Reply:
Resolved. Following the implemetation in Windows extension, i.e.
1.	Log error details in Configure Patching substatus
2.	Continue with rest of the Installation

NOTE: Windows does not mark the uber-operation as failed, if configure patching fails. Following the same approach in Linux. 

Q: Do we need to change this to mark the uber-operation as failed?